tensorflow/tools/ci_build/builds/cmake.sh

```diff
@@ -16,7 +16,54 @@
 
 set -e
 
+
+# Determine the number of cores, for parallel make.
+N_JOBS=$(grep -c ^processor /proc/cpuinfo)
+if [[ -z ${N_JOBS} ]]; then
+  # The Linux way didn't work. Try the Mac way.
+  N_JOBS=$(sysctl -n hw.ncpu)
+fi
+if [[ -z ${N_JOBS} ]]; then
+  N_JOBS=1
+  echo ""
+  echo "WARNING: Failed to determine the number of CPU cores. "\
+"Will use --jobs=1 for make."
+fi
+
+echo ""
+echo "make will use ${N_JOBS} concurrent job(s)."
+echo ""
+
+
+# Build and install protobuf.
+PROTOBUF_VERSION="3.0.0-beta-2"
+PROTOBUF_DOWNLOAD_DIR="/tmp/protobuf"
+
+mkdir "${PROTOBUF_DOWNLOAD_DIR}"
+pushd "${PROTOBUF_DOWNLOAD_DIR}"
+curl -fSsL -O https://github.com/google/protobuf/releases/download/v$PROTOBUF_VERSION/protobuf-cpp-$PROTOBUF_VERSION.tar.gz
+tar zxf protobuf-cpp-$PROTOBUF_VERSION.tar.gz
+cd protobuf-$PROTOBUF_VERSION
+./autogen.sh
+./configure
+make --jobs=${N_JOBS}
+sudo make install
+make clean
+sudo ldconfig
+popd
+rm -rf "${PROTOBUF_DOWNLOAD_DIR}"
+
+
+# Run TensorFlow cmake build.
+# Clean up, because certain modules, e.g., highwayhash, seem to be sensitive
+# to state.
+rm -rf build
+
 mkdir -p build
-cd build
+pushd build
 cmake -DCMAKE_BUILD_TYPE=Release ../tensorflow/contrib/cmake
+
+# TODO(mmry, caisq): Use --jobs>1 after fixing dependency issues.
```

Comment:
Nit: "TODO(mrry, caisq):" (although we might be able to fix the dependency issue in this PR, by modifying https://github.com/tensorflow/tensorflow/blob/f49afa02c521495d5f4fea59a6a58d38c75bb596/tensorflow/contrib/cmake/tf_models.cmake#L74 to refer to the `_kernels` target rather than the `_ops` target.